staging/src/k8s.io/component-base/metrics/testutil/metrics.go

```diff
@@ -178,3 +183,139 @@ func ValidateMetrics(metrics Metrics, metricName string, expectedLabels ...strin
 	}
 	return nil
 }
+
+// Histogram wraps prometheus histogram DTO (data transfer object)
+type Histogram struct {
+	histogram *dto.Histogram
+}
+
+// GetHistogramFromDefaultGatherer gathers a metric from k8s.io/component-base/metrics/legacyregistry.DefaultGatherer.
+// Used only for testing purposes where we need to gather metrics directly from a running binary (without metrics endpoint).
+func GetHistogramFromDefaultGatherer(metricName string) (Histogram, error) {
```

Comment:
We should just pass a Gatherer into this function and invoke that, that'll allow us to shed the legacyregistry dep.

Reply:
Done